### DISCUSSION TITLE
[not a security vulnerability] qgis-deployment-cli flagged as trojan agent

### DISCUSSION THREAD

**Discussion by Jean-Roc** [Guts/qgis-deployment-cli#570] [General]
Hi, the following is not an issue but an example of a failed attempt at using qlc in a french administration : 
our IT dept is rejecting the use of qgis-deployment based on this automated [report](https://hybrid-analysis.com/sample/fb3752655e8f0f8736839de12431f9da8fa212ae9a6e0577fa07cd5b873a0752/67110549ebdbf67dcd03923d) labeling the exe file as a "_Trojan Agent_". In my opinion, most (if not all) of the listed points in the "_Informative_" section are expected and the "_Suspicious_" clearly false positive.

**Comment by Guts** [comment]
Thanks @Jean-Roc for raising up this report.

The WMI use (through https://github.com/jcarswell/pyad) is related to the feature that look into Active Directory to deploy profiles conditionally to the end-user's groups.

Since QDT is open source, the code can be audited to make ITself count there is no trojan behavior inside.

Furthermore, we already passed through IT checks by:

- adding a bunch of security checks which are run for every commit
- writing a quite complete security policy: https://github.com/Guts/qgis-deployment-cli?tab=security-ov-file

We also recommend to sign the executable with an internal one: https://guts.github.io/qgis-deployment-cli/guides/howto_windows_sign_executable.html.

Is it possible to discuss the sticking points? We could, for example, consider disabling active directory-related functionalities with a configuration flag.

**Comment by Jean-Roc** [comment]
Hi, thank you very much for those details and your proposal, I'm waiting on their feedback to see if our IT team accepts to get past the _"trojan"_ label.

**Comment by Guts** [comment]
Hi @Jean-Roc 

FYI I've uploaded the latest QDT binary to https://virusscan.jotti.org:

![image](https://github.com/user-attachments/assets/9796ae7a-ba00-4d7b-b0e4-f73202eb9082)

**Comment by Jean-Roc** [comment]
Thank you for this update ! I'm sharing it with my colleagues

**Reply by Guts** [reply]
Hi @Jean-Roc,

Any news or feedback?

**Reply by Jean-Roc** [reply]
Hi, sorry but no feedback and none expected for a while, our IT colleagues on this matter have been fully outsourced.

**Comment by Guts** [comment]
@Jean-Roc I suggest to close here since there is apparently no discussion...

**Comment by Jean-Roc** [comment]
yes indeed
